core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -191,6 +191,13 @@ class LogCleaner(initialConfig: CleanerConfig,
     */
   override def reconfigure(oldConfig: KafkaConfig, newConfig: KafkaConfig): Unit = {
     config = LogCleaner.cleanerConfig(newConfig)
+
+    val maxIoBytesPerSecond = config.maxIoBytesPerSecond;
+    if (maxIoBytesPerSecond != oldConfig.logCleanerIoMaxBytesPerSecond) {
+      info(s"Updating logCleanerIoMaxBytesPerSecond: $maxIoBytesPerSecond")
+      throttler.updateDesiredRatePerSec(maxIoBytesPerSecond)
+    }
```

Comment:
Could we also add a test for `LogCleaner` to make sure after `reconfigure`, the throttler will get the updated config?

Reply:
I’ve added a test for that. I could not find reconfigure test that would be a good reference that I could imitate, so this is a test in my own way.
https://github.com/apache/kafka/pull/12296/commits/1a574d67ad1685ce776bb45896fe6dd72992dbef

Reply:
Test is good. Thanks.